### PR TITLE
build: fix eslint issues

### PR DESCRIPTION
**Pull request by noomorph**
I wasn't able to run ESLint of files due to various errors:

* `Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/util/glob-util' is not defined by "exports"`
* `Failed to load plugin 'import' declared in '.eslintrc » eslint-config-airbnb`
* `require() of ES modules is not supported" when package.json has "type":"module"`

After a battle with dependencies and configs, `package-lock.json` and so on, I also noticed a few issues around the code itself, and fixed them too.

That's the result. 

Thanks for the project. Honestly, it feels like you don't have time to keep it up-to-date, but anyway it still looks like the best out-of-the-box solution currently on GitHub.

P. S. I had to lock `.nvmrc` to Node 14.x, in fact. When running with Node 16.x, I see too many errors – so I'd suggest you to review that separately if you'd like to unlock it further.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,54 +34,56 @@
     "npm": ">=6.0.0"
   },
   "dependencies": {
-    "prop-types": "^15.7.2",
+    "jest-environment-node": "^27.5.1",
```
Comment (enuchi) [inline package.json:37]: can you explain what `jest-environment-node` is needed for (and does it need to be a dependency or dev dependency)?

**Comment by enuchi** [comment]
Thanks for opening this @noomorph. You're right, it's hard to keep this up to date. The main issue is some of the webpack plugins needed are not maintained so it's not easy to move to webpack 5. @kmalakoff has an open PR for this but have not had a chance to look (#103).

Thanks for pinning the .nvmrc file -- ideally should be v16. Do you know what kind of errors it was showing?

File: package.json (hunk 1 above)
Comment (noomorph) [reply package.json:37]: Looks like something accidental or automatic, I'll check.

Comment (noomorph) [reply package.json:37]: Yes, looks like automatic issue:

image

Comment (noomorph) [reply package.json]: Hmm... okay, I moved it to `devDependencies`

**Comment by noomorph** [comment]
@enuchi looks like it comes from `npm v8`. It seems to be stricter:

image

**Comment by noomorph** [comment]
Okay, I found the reason. Upgrading to `eslint-config-airbnb-base@15.0.0` helped.

**Comment by enuchi** [comment]
Thanks for the work on this @noomorph, much appreciated

**Pull request merged**